src/Core/Services/UsernameGenerationService.cs

```diff
@@ -0,0 +1,250 @@
+            }
+            else
+            {
+                generatedString = options.EmailWebsite;
+            }
+
+            return generatedString + "@" + catchAllEmailDomain;
+        }
+
+        private async Task<string> GenerateForwardedEmailAliasAsync(UsernameGenerationOptions options)
+        {
+            options.Merge(_defaultOptions);
+
+            switch (options.ServiceType)
+            {
+                case ForwardedEmailServiceType.AnonAddy:
+                    if (string.IsNullOrWhiteSpace(options.AnonAddyApiAccessToken) || string.IsNullOrWhiteSpace(options.AnonAddyDomainName))
+                    {
+                        return DefaultGenerated;
+                    }
+                    return await GetAnonAddyUsername(options.AnonAddyApiAccessToken, options.AnonAddyDomainName);
+
+                case ForwardedEmailServiceType.FirefoxRelay:
+                    if (string.IsNullOrWhiteSpace(options.FirefoxRelayApiAccessToken))
+                    {
+                        return DefaultGenerated;
+                    }
+                    return await GetFirefoxRelayUsername(options.FirefoxRelayApiAccessToken);
+
+                case ForwardedEmailServiceType.SimpleLogin:
+                    if (string.IsNullOrWhiteSpace(options.SimpleLoginApiKey))
+                    {
+                        return DefaultGenerated;
+                    }
+                    return await GetSimpleLoginUsername(options.SimpleLoginApiKey);
+                default:
+                    return DefaultGenerated;
+            }
+        }
+
+        private async Task<string> GetFirefoxRelayUsername(string apiToken)
+        {
+            var url = "https://relay.firefox.com/api/v1/relayaddresses/";
+
+            return await _apiService.GetUsernameFromFirefoxRelay(url, apiToken);
+        }
+
+        private async Task<string> GetSimpleLoginUsername(string apiToken)
+        {
+            var url = "https://app.simplelogin.io/api/alias/random/new";
+
+            return await _apiService.GetUsernameFromSimpleLogin(url, apiToken);
+        }
+
+        private async Task<string> GetAnonAddyUsername(string apiToken, string domain)
+        {
+            var url = "https://app.anonaddy.com/api/v1/aliases";
+
+            return await _apiService.GetUsernameFromAnonAddy(url, apiToken, domain);
+        }
+
+        private async Task<string> RandomStringAsync(int length)
+        {
+            var str = "";
+            var charSet = "abcdefghijklmnopqrstuvwxyz1234567890";
+
+            for (var i = 0; i < length; i++)
+            {
+                var randomCharIndex = await _cryptoService.RandomNumberAsync(0, charSet.Length - 1);
+                str += charSet[randomCharIndex];
+            }
+
+            return str;
+        }
+
+        private string Capitalize(string str)
+        {
+            return str.First().ToString().ToUpper() + str.Substring(1);
```

Comment:
if `str` is `null` or empty this will throw an exception, and don't use `First().ToString().ToUpper()` , strings are array of char so `str[0].ToUpper()` is the same as all the other